nexus-tweaks.user.js

```diff
@@ -1144,6 +1144,264 @@ promiseList.push((async () => {
+            ]
+        },
+        "Monster Hunting" : {
+            "Crystal Guardian" : ["Shardbreaker", "Crystal Smasher", "Entropy Incarnate"],
+            "Great Wyrm" : ["Wyrm Hunter", "Wyrm Slayer", "Wyrmsbane"],
+            "Maeval" : ["Stalker of Death", "Hunter of Death", "Foe of Death"]
+        }
+    }
+
+    function classifyBadges(badgesText) {
+        const badgeTexts = badgesText.trim().split(",").map(str => str.trim())
+        let classifiedBadges = {}
+
+        for (let bT of badgeTexts) {
+            let found = false
+            for (let firstLevel in Badges) {
+                for (let secondLevel in Badges[firstLevel]) {
+                    if (Badges[firstLevel][secondLevel].includes(bT)) {
+                        if (!(firstLevel in classifiedBadges)) classifiedBadges[firstLevel] = {}
+                        if (!(secondLevel in classifiedBadges[firstLevel])) classifiedBadges[firstLevel][secondLevel] = []
+                        classifiedBadges[firstLevel][secondLevel].push(bT)
+                        found = true
+                        continue
+                    }
+                }
+                if (found) continue
+            }
+            if (!found) {
+                if (!classifiedBadges.Others) classifiedBadges.Others = []
+                classifiedBadges.Others.push(bT)
+            }
+        }
+
+        return classifiedBadges
+    }
+
+    function printClassifiedBadges(classifiedBadges, badgesNode) {
+        function L1(text, parent) {
+            const categoryNode = parent.appendChild(document.createElement('div'))
+            categoryNode.style.backgroundColor = "#dddddd"
+            const titleNode = categoryNode.appendChild(document.createElement('div'))
+            const contentNode = categoryNode.appendChild(document.createElement('div'))
+            titleNode.style.fontWeight = "900"
+            titleNode.textContent = text
+            titleNode.title = "Click to expand/collapse"
+            titleNode.onclick = () => { if (contentNode.style.display != "none") contentNode.style.display = "none"; else contentNode.style.display = "initial" }
+            return contentNode
+        }
+        function L2(text, parent) {
+            const newNode = document.createElement('div')
+            newNode.style.backgroundColor = "#eeeeee"
+            newNode.style.fontWeight = "normal"
+            newNode.textContent = text
+            return parent.appendChild(newNode)
+        }
+        function L3(text, parent) {
+            const newNode = document.createElement('div')
+            newNode.style.backgroundColor = "#ffffff"
+            newNode.style.fontWeight = "normal"
+            newNode.textContent = text
+            return parent.appendChild(newNode)
+        }
+        function orderObject(unordered, ordering) {
```

Comment:
An inline comment documenting the parameters of this function, their types, and how it functions would not go amiss.